UnitsNet.Tests/GeneratedCode/TestsBase/AccelerationTestsBase.g.cs

```diff
@@ -268,73 +309,41 @@ public void As_SIUnitSystem_ThrowsArgumentExceptionIfNotSupported()
-            Assert.Equal(AccelerationUnit.MillimeterPerSecondSquared, millimeterpersecondsquaredQuantity.Unit);
-
-            var millistandardgravityQuantity = meterpersecondsquared.ToUnit(AccelerationUnit.MillistandardGravity);
-            AssertEx.EqualTolerance(MillistandardGravityInOneMeterPerSecondSquared, (double)millistandardgravityQuantity.Value, MillistandardGravityTolerance);
-            Assert.Equal(AccelerationUnit.MillistandardGravity, millistandardgravityQuantity.Unit);
-
-            var nanometerpersecondsquaredQuantity = meterpersecondsquared.ToUnit(AccelerationUnit.NanometerPerSecondSquared);
-            AssertEx.EqualTolerance(NanometersPerSecondSquaredInOneMeterPerSecondSquared, (double)nanometerpersecondsquaredQuantity.Value, NanometersPerSecondSquaredTolerance);
-            Assert.Equal(AccelerationUnit.NanometerPerSecondSquared, nanometerpersecondsquaredQuantity.Unit);
+            var conversionFactor = GetConversionFactor(unit);
+            AssertEx.EqualTolerance(conversionFactor.UnitsInBaseUnit, (double)converted.Value, conversionFactor.Tolerence);
+            Assert.Equal(unit, converted.Unit);
+        }
 
-            var standardgravityQuantity = meterpersecondsquared.ToUnit(AccelerationUnit.StandardGravity);
-            AssertEx.EqualTolerance(StandardGravityInOneMeterPerSecondSquared, (double)standardgravityQuantity.Value, StandardGravityTolerance);
-            Assert.Equal(AccelerationUnit.StandardGravity, standardgravityQuantity.Unit);
+        [Theory]
+        [MemberData(nameof(UnitTypes))]
+        public void ToUnit_WithSameUnits_AreEqual(AccelerationUnit unit)
+        {
+            var quantity = Acceleration.From(3.0, unit);
+            var toUnitWithSameUnit = quantity.ToUnit(unit);
+            Assert.Equal(quantity, toUnitWithSameUnit);
         }
 
-        [Fact]
-        public void ToUnit_WithBaseUnit_ReturnsQuantityWithBaseUnit()
+        [Theory]
+        [MemberData(nameof(UnitTypes))]
+        public void ToUnit_FromNonBaseUnit_NoException(AccelerationUnit unit)
         {
-            var quantityInBaseUnit = Acceleration.FromMetersPerSecondSquared(1).ToUnit(Acceleration.BaseUnit);
-            Assert.Equal(Acceleration.BaseUnit, quantityInBaseUnit.Unit);
+            // See if there is a unit available that is not the base unit.
+            var fromUnit = Acceleration.Units.FirstOrDefault(u => u != Acceleration.BaseUnit && u != AccelerationUnit.Undefined);
+
+            // If there is only one unit for the quantity, we must use the base unit.
+            if(fromUnit == AccelerationUnit.Undefined)
+                fromUnit = Acceleration.BaseUnit;
+
+            var quantity = Acceleration.From(3.0, fromUnit);
+            var converted = quantity.ToUnit(unit);
+            // TODO: Meaningful check possible?
```

Comment:
Can't we assert that the new quantity has `Unit == unit`?
If so, I would rename this method `ToUnit_ReturnsQuantityWithGivenUnit` unless there is already an equivalent test for that. No exception is an implicit assertion in tests.